Better SDK34 support (#88)
MindscapeHQ/raygun4android: README.md

```gradle
dependencies {
    ...

>     implementation 'com.raygun:raygun4android:4.0.1'

Reviewer: What's the reason behind this change?

Author: I want to tell people that at this stage - until anything else is published - 4.0.1 is the stable released. `4.+` might just unintendend suck in a 4.1.0 alpha/snapshot when I publish it.

For an actual usage example, check out the sample application in the **app** module of this project

## Raygun and ProGuard/R8

### General

> ProGuard and R8 are tools for obfuscation, class file shrinking, optimizing and pre-verifying. When enabling ProGuard in a native Android application that also uses Raygun, the obfuscation feature requires a bit of attention. By default, your obfuscated class and method names will show up in the stacktraces of exception/error reports submitted to Raygun. This makes the stacktraces difficult to read when looking into the cause of the issues.

Reviewer: Is it no longer free or is R8 not free bu ProGuard is?

Author: The `free` comment doesn't really make sense in this context and is probably still in the doc from before I even took it over :). It comes with Android Studio as-is anyway and is now - depending on what AS folks use - either ProGuard or R8.

This function gets called from within the ```android {...}``` block of the Gradle file at each build in Android Studio and creates the appropriate parameterised task to notify the Raygun backend of your app's deployment.

> ## Advanced features

Reviewer: We mention Advanced features in most of our provider's documentation. For consistency and familiarity across the providers, we should keep this in place

Author: Fixed

### Storing crash reports on the device

> If the device can't connect, Raygun4Android will save the crash report to storage. At the next start of the application (and therefore the provider) it will check if the internet is now available, and if it is, send the cached messages. A maximum of 64 messages will be cached and you can change the amount by calling:

Reviewer: ```suggestion
If the device can't connect, Raygun4Android will save the crash report to the device storage. At the next start of the application, (along with the provider) it will check if the internet is now available. If it is, send the cached messages. A maximum of 64 messages will be cached and you can change the amount by calling:
```

> If the device can't connect, Raygun4Android will save the crash report to storage. At the next start of the application (and therefore the provider) it will check if the internet is now available, and if it is, send the cached messages. A maximum of 64 messages will be cached and you can change the amount by calling:

Author: If the storage limit (with the default of 64) is reached, it will just not write more unless there is space. If that's a good way of doing it, I'm not certain - but it was at the time aligning how Mitchell had written the iOS provider, so we tried to keep the behaviour in line.

I think the default of 64 is not really that reasonable anymore these days either - we should consider to change that maybe. There's also https://github.com/MindscapeHQ/raygun4android/issues/49 - that goes into that direction of "we should really consider changing that" as well.

> If the device can't connect, Raygun4Android will save the crash report to storage. At the next start of the application (and therefore the provider) it will check if the internet is now available, and if it is, send the cached messages. A maximum of 64 messages will be cached and you can change the amount by calling:

Author: Just pushed an updated with a note on the current behaviour.

* Is there an example app?

  Yup - clone this repository, then run the **app** module of the project.

> * I'm not seeing errors in the dashboard.

Reviewer: ```suggestion
* I'm not seeing errors in Raygun Crash Reporting.
```

Reviewer: Because referring to the Crash Reporting product as a dashboard is not something we do anymore. We have an actual product called Dashboards so this will be confusing to keep!

  Raygun4Android outputs Logcat messages - look for the the logcat tag **Raygun4Android**. HTTP Status 403 will indicate an invalid API key, 400 a bad message, and 202 will indicate received successfully.

* My build fails with `Default interface methods are only supported starting with Android N (--min-api 24)`. Why is that?

>   Raygun4Android uses Timber for internal logging. This requires some language features that are only available with Java 8. Make sure that your project using the library has set compilation compatibility to Java 8 as the minimum

Reviewer: ```suggestion
  Raygun4Android uses Timber for internal logging. This requires some language features that are only available with Java 8. Make sure that your project, using the library, has set compilation compatibility to Java 8 as the minimum
```

* The library logs an error message about a not found class: Rejecting re-init on previously-failed class java.lang.Class<android.support.v4.app.JobIntentService$JobServiceEngineImpl>

  The message above stems from certain versions of the Android support libraries. JobServiceEngineImpl is part of Android Oreo (8, SDK 26) and newer only. The support library catering for supporting services on earlier versions of Android runs internal checks for which implementation is available to it. As part of the checks, it outputs the message as an informational feedback only.

* Timber Lint warnings get exposed to my app even though I don't (want to) use Timber.

>   The solution for the time being is to disable linting for those specific lint warning in your app:

Reviewer: Can we say for production? because it will helpful for the developer locally but not in production towards the app users

Author: Just to clarify - the users of an app will never see this.

What happens is:

- `Raygun4Android:provider` uses Timber (a library for better logging on the device's logcat)
- Timber comes with its own set of Android (Studio) Lint rules and they're opinionated.
- Due to how Android's toolchain works, the linting rules are passed "downstream" to any modulde/app that uses a library (our `:provider`) that uses Timber. In this case to `Raygun4Android:app`.
- So, anything that *uses* our `:provider` gets the Timber linting rules passed through to it, regardless if that app is using Timber itself or not (many apps do anyway).
- Effectively that means that an app developer not using Timber in their app sees linting rules and might be confused. From here the options for them are to switch off linting in their IDE completely or ignore the lint rules in their code etc - but the cleanest solution is currently to add this disable statement into the build.

So, this is not inherently a `debug` or `release` issue. In their app, they should do this anyway if they don't use Timber themselves.
